ui/lib/services/project_service.dart

```diff
@@ -52,6 +45,14 @@ class ProjectService {
     });
   }
 
+  /// Returns all projects that match the [fragment] with their (PURL-based) reference.
+  /// Supports "projects_fragment/name_fragment" as fragment specification.
+  Future<List<Project>> findProjects([String? fragment]) => _execute(() async {
+        final projects = await _client.findProjectsBySearchFragment(fragment);
+        log('Searching projects matching "$fragment"');
```

Comment:
When not providing a fragment, this will log _Searching projects matching "null"_

Perhaps somehow indicate explicitly that no fragment was provided?